plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/snowflake/TestIcebergSnowflakeCatalogConnectorSmokeTest.java

```diff
@@ -0,0 +1,768 @@
+    }
+
+    @Test
+    @Override
+    public void testTableChangesFunction()
+    {
+        assertThatThrownBy(super::testTableChangesFunction)
+                .hasMessageMatching("Snowflake managed Iceberg tables do not support modifications");
+    }
+
+    @Test
+    @Override
+    public void testRowLevelDeletesWithTableChangesFunction()
+    {
+        assertThatThrownBy(super::testRowLevelDeletesWithTableChangesFunction)
+                .hasMessageMatching("Snowflake managed Iceberg tables do not support modifications");
+    }
+
+    @Test
+    @Override
+    public void testCreateOrReplaceWithTableChangesFunction()
+    {
+        assertThatThrownBy(super::testCreateOrReplaceWithTableChangesFunction)
+                .hasMessageMatching("Snowflake managed Iceberg tables do not support modifications");
+    }
+
+    @Test
+    public void testNation()
+    {
+        assertQuery("SELECT count(*) FROM " + TpchTable.NATION.getTableName(), "VALUES 25");
+        assertTableColumnNames(TpchTable.NATION.getTableName(), "nationkey", "name", "regionkey", "comment");
+    }
+
+    @Test
+    public void testListTables()
+    {
+        assertThat(computeActual("SHOW TABLES").getMaterializedRows().stream()
+                .map(row -> row.getField(0))
+                .toList()).contains(TpchTable.REGION.getTableName(), TpchTable.NATION.getTableName());
+    }
+
+    @Test
+    public void testRegion()
+    {
+        assertQuery("SELECT count(*) FROM " + TpchTable.REGION.getTableName(), "VALUES 5");
+        assertTableColumnNames(TpchTable.REGION.getTableName(), "regionkey", "name", "comment");
+        assertQuery("SELECT name FROM " + TpchTable.REGION.getTableName(), "VALUES ('AFRICA'), ('AMERICA'), ('ASIA'), ('EUROPE'), ('MIDDLE EAST')");
+    }
+
+    @Test
+    public void testSetTableComment()
+    {
+        assertThatThrownBy(() -> assertUpdate("COMMENT ON TABLE " + TpchTable.REGION.getTableName() + " is 'my-table-comment'"))
+                .hasMessage("Snowflake managed Iceberg tables do not support modifications");
+    }
+
+    @Test
+    public void testSetViewComment()
+    {
+        assertThatThrownBy(() -> assertUpdate("COMMENT ON VIEW temp_view is 'my-table-comment'"))
+                .hasMessageMatching("line 1:1: View '.*' does not exist");
+    }
+
+    @Test
+    public void testSetViewColumnComment()
+    {
+        assertThatThrownBy(() -> assertUpdate("COMMENT ON COLUMN temp_view.col1 is 'my-column-comment'"))
+                .hasMessageMatching(".*Table does not exist: .*temp_view");
+    }
+
+    @Test
+    public void testSetMaterializedViewColumnComment()
+    {
+        assertThatThrownBy(() -> assertUpdate("COMMENT ON COLUMN temp_view.col1 is 'my-column-comment'"))
+                .hasMessageMatching(".*Table does not exist: .*temp_view");
+    }
+
+    @Test
+    public void testDropTable()
+    {
+        assertThatThrownBy(() -> assertUpdate("DROP TABLE " + TpchTable.REGION.getTableName()))
```

Comment:
I don't think trying to modify TPCH tables in shared service is good idea. It may lead to accidental change when we change Snowflake catalog in the future. We faced such issue in BigQuery connector in the past.

Reply:
That could very well happen in future when Snowflake catalog supports modification and we forget to modify/remove this test. I'll remove this test for now.